packages/edit-site/src/components/block-editor/index.js

```diff
@@ -51,28 +58,252 @@ const LAYOUT = {
 	alignments: [],
 };
 
+const FOCUSABLE_ENTITIES = [ 'wp_template_part', 'wp_navigation' ];
+
 export default function BlockEditor() {
-	const { setIsInserterOpened } = useDispatch( editSiteStore );
-	const { storedSettings, templateType, canvasMode, hasPageContentFocus } =
-		useSelect(
-			( select ) => {
-				const {
-					getSettings,
-					getEditedPostType,
-					getCanvasMode,
-					hasPageContentFocus: _hasPageContentFocus,
-				} = unlock( select( editSiteStore ) );
-
-				return {
-					storedSettings: getSettings( setIsInserterOpened ),
-					templateType: getEditedPostType(),
-					canvasMode: getCanvasMode(),
-					hasPageContentFocus: _hasPageContentFocus(),
-				};
-			},
-			[ setIsInserterOpened ]
+	const { templateType, hasPageContentFocus } = useSelect( ( select ) => {
+		const { getEditedPostType, hasPageContentFocus: _hasPageContentFocus } =
+			unlock( select( editSiteStore ) );
+
+		return {
+			templateType: getEditedPostType(),
+			hasPageContentFocus: _hasPageContentFocus(),
+		};
+	}, [] );
+
+	const settings = useSiteEditorSettings( templateType );
```

Comment:
This one seems better in the "specific" providers (navigation, default). It's not clear to me why we're retrieving the settings here.

Reply:
Just legacy code. Now it's a hook we can call it in the providers rather than prop drill 👍

Reply:
I moved the settings retrieval to the specific components.